### PR TITLE
chore: fix incorrect initial assignment to contains_loop_var

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes https://github.com/lfortran/lfortran/pull/4019#issuecomment-2118240991

### CODE REVIEW

**Comment by Pranavchiku** [comment]
MRE:

```fortran
program main
integer, parameter :: n = 3
real:: v(n), a(n,n), b(n,n)
integer :: i,j

v = [(i,i=1,n)]
a = diag_rsp(v)
b = reshape([((merge(i,0,i==j), i=1,n), j=1,n)], [n,n])
print *, "all(a==b)", all(a==b)
contains
function diag_rsp(v) result(res)
    real, intent(in) :: v(:)
    real :: res(size(v),size(v))
    integer :: i
    res = 0
    do i = 1, size(v)
      res(i,i) = v(i)
    end do
end function diag_rsp
end program
```

```console
% lfortran a.f90           
all(a==b) False
 % gfortran a.f90 && ./a.out
 all(a==b) T
```

**Comment by Pranavchiku** [comment]
The trouble is we do not have compile time implementation of majority of array intrinsic and thus I had to skip `IntrinsicArrayFunction`

File: tests/reference/asr-arrays_06-9c8a08b.stdout [hunk 1]
```diff
@@ -111,29 +111,15 @@
...
+                        (ArrayConstant
+                            24
+                            [2, 4, 6, 8, 10, 12]
                             (Array
                                 (Integer 4)
                                 [((IntegerConstant 1 (Integer 4))
-                                ())]
-                                DescriptorArray
+                                (IntegerConstant 6 (Integer 4)))]
+                                FixedSizeArray
```
Comment (certik) [inline tests/reference/asr-arrays_06-9c8a08b.stdout:121]: I think this change does not seem correct: we want ArrayConstructor here, with the symbolic code, and then in it we want a compile time value of ArrayConstant.

Comment (Pranavchiku) [reply tests/reference/asr-arrays_06-9c8a08b.stdout:121]: I’ll check this in a while, this won’t be difficult to fix, small change will do the job.

Comment (Pranavchiku) [reply tests/reference/asr-arrays_06-9c8a08b.stdout]: The latest commit does the job, now we keep `implied do loop` and set value as array constant. In this case, we call `make_ArrayConstructor_util` which checks if it is compile time then creates an ArrayConstant and loses symbolic value. It is not related to this PR, fix is minor though. How about doing it in subsequent PRs?

Comment (Pranavchiku) [reply tests/reference/asr-arrays_06-9c8a08b.stdout]: MRE:

```fortran
program main
implicit none
real, dimension(3) :: a
integer :: i
a = [(i, i = 1, 3)]
print *, a
end program
```

**Review by certik — APPROVED** [review]
Ok, this is fine to merge this PR if you can fix the last part after that. Right now there is a regression where it removes ImpliedDoLoop from ASR and replaces it with a constant.

I'll leave it for you to merge.

File: tests/reference/asr-arrays_06-9c8a08b.stdout (hunk 1 above)
Comment (certik) [reply tests/reference/asr-arrays_06-9c8a08b.stdout]: If you can do it, then we can fix this in a subsequent PR.

**Comment by Pranavchiku** [comment]
I am enabling auto merge and working on fix.

**Pull request merged**